sdk/redisenterprise/Azure.ResourceManager.RedisEnterprise/src/Generated/AccessPolicyAssignmentData.cs

```diff
@@ -0,0 +1,94 @@
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("\"foo\"")</term>
+        /// <description>Creates a payload of "foo".</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromObjectAsJson(new { key = "value" })</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// <item>
+        /// <term>BinaryData.FromString("{\"key\": \"value\"}")</term>
+        /// <description>Creates a payload of { "key": "value" }.</description>
+        /// </item>
+        /// </list>
+        /// </para>
+        /// </summary>
+        private IDictionary<string, BinaryData> _serializedAdditionalRawData;
+
+        /// <summary> Initializes a new instance of <see cref="AccessPolicyAssignmentData"/>. </summary>
+        public AccessPolicyAssignmentData()
+        {
+        }
+
+        /// <summary> Initializes a new instance of <see cref="AccessPolicyAssignmentData"/>. </summary>
+        /// <param name="id"> The id. </param>
+        /// <param name="name"> The name. </param>
+        /// <param name="resourceType"> The resourceType. </param>
+        /// <param name="systemData"> The systemData. </param>
+        /// <param name="provisioningState"> Current provisioning status of the access policy assignment. </param>
+        /// <param name="accessPolicyName"> Name of access policy under specific access policy assignment. Only "default" policy is supported for now. </param>
+        /// <param name="user"> The user associated with the access policy. </param>
+        /// <param name="serializedAdditionalRawData"> Keeps track of any properties unknown to the library. </param>
+        internal AccessPolicyAssignmentData(ResourceIdentifier id, string name, ResourceType resourceType, SystemData systemData, RedisEnterpriseProvisioningStatus? provisioningState, string accessPolicyName, AccessPolicyAssignmentPropertiesUser user, IDictionary<string, BinaryData> serializedAdditionalRawData) : base(id, name, resourceType, systemData)
+        {
+            ProvisioningState = provisioningState;
+            AccessPolicyName = accessPolicyName;
+            User = user;
+            _serializedAdditionalRawData = serializedAdditionalRawData;
+        }
+
+        /// <summary> Current provisioning status of the access policy assignment. </summary>
+        public RedisEnterpriseProvisioningStatus? ProvisioningState { get; }
+        /// <summary> Name of access policy under specific access policy assignment. Only "default" policy is supported for now. </summary>
+        public string AccessPolicyName { get; set; }
+        /// <summary> The user associated with the access policy. </summary>
+        internal AccessPolicyAssignmentPropertiesUser User { get; set; }
+        /// <summary> The object ID of the user. </summary>
+        public string UserObjectId
```

Comment:
What's the format of this property value?
- If it's a GUID, use type [Guid](https://learn.microsoft.com/en-us/dotnet/api/system.guid?view=net-6.0). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/maps/Azure.ResourceManager.Maps/src/autorest.md?plain=1#L25).
- If it's a resource identifier, use type [ResourceIdentifier](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/ResourceIdentifier.cs). See example [here](https://github.com/Azure/azure-sdk-for-net/blob/bd953fc7a97bf617c892ce6314b50126b4021a4f/sdk/search/Azure.Search.Documents/src/autorest.md?plain=1#L106-L115).

Reply:
I think it should be `AccessPolicyAssignmentProperties.user.objectId`, per spec: https://github.com/Azure/azure-rest-api-specs/blob/e38a6d8c499aa984447d6c3ed8d686611405bf55/specification/redisenterprise/resource-manager/Microsoft.Cache/preview/2024-09-01-preview/redisenterprise.json#L2278C14-L2278C22

Reply:
done this change locally and still see string

Reply:
try `AccessPolicyAssignmentPropertiesUser.objectId: -|uuid`.
![image](https://github.com/user-attachments/assets/dbe9a404-8923-4091-a074-4b541895a5f4)